### PR TITLE
[RNMobile] Fix a11y issue in Cover block

### LINKED ISSUE #2699: [iOS][Accessibility] Cover Block: Fix selecting background image with VoiceOver on
When first adding a Cover block and selecting it, it takes you into adding a paragraph block rather than giving you the option to add an image immediately as the block does when Voice Over is off. Here’s a video of the issue:

https://d.pr/v/jllfUq

If you try to work around the issue with the Cover block directing you to add a paragraph block inside the cover block (instead of an image) using the block navigation toolbar to “Navigate Up” in the block hierarchy to select the cover block itself, there is an image icon that appears in the upper right of the cover block that can be selected. Voice Over mentions the button when describing the contents of the block (button, image), but when the button is actually selected, it does not say anything. This button is the button that gives you the option to select an image or video for use with the cover block, so the fact it isn’t spoken when selected is a bit of a problem. Here’s a video of this issue:

https://d.pr/v/sQj2Me

**To Reproduce**
1. With VoiceOver enabled
2. Open the WordPress app (or the demo app)
3. Create a new post
4. Open the block inserter
5. Select and add a cover block
6. Select the cover block
7. Notice you're invited to edit text inside a paragraph instead of picking between adding a background image or editing text

**Expected behavior**
Selecting a cover block which does not have a background set should invite the user to pick a background image first.

**Smartphone (please complete the following information):**
 - Device: iPhone
 - OS: all iOS versions

**Additional context**
Reported by Galen, noticed in a user ticket: 3316181-zen

### PR DESCRIPTION
**Pull request by lukewalczak**
## Description

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2699

That PR is introducing fixes for a11y related to Cover block such as:

* selecting the Cover block gives you the option to choose an image
* selecting a edit button over the Cover with an image (to choose different image or video, see the screenshot below) gives user hints about its functionality

Screenshot 2021-02-02 at 11 48 55

## How has this been tested?

1. Enable VoiceOver
2. Add new post
3. Add `Cover` block
4. Select a `Cover` block

** Expect picker appears and you can choose one of the options **

5. Add a photo
6. Select a edit button over the image

** Expect to hear the hints about buttons functionality **

## Types of changes

Bug fix / Accessibility improvements

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/cover/edit.native.js [hunk 1]
```diff
@@ -106,9 +107,36 @@ const Cover = ( {
...
+
+		return () => {
+			AccessibilityInfo.removeEventListener(
+				'screenReaderChanged',
+				handleScreenReaderToggled
+			);
+		};
+	}, [] );
+
+	const handleScreenReaderToggled = ( enabled ) => {
```
Comment (dratwas) [inline packages/block-library/src/cover/edit.native.js:136]: Do we need that function? Can't we simply use the `setIsScreenReaderEnabled` ?

**Review by dratwas — APPROVED** [review]
Tested it and works as expected, however, I left one small comment :)

File: packages/block-library/src/cover/edit.native.js [hunk 2]
```diff
@@ -106,9 +107,32 @@ const Cover = ( {
...
 	} = attributes;
+	const [ isScreenReaderEnabled, setIsScreenReaderEnabled ] = useState(
+		false
+	);
 
 	const CONTAINER_HEIGHT = minHeight || COVER_DEFAULT_HEIGHT;
 
+	useEffect( () => {
+		// sync with local media store
+		mediaUploadSync();
```
Comment (dratwas) [inline packages/block-library/src/cover/edit.native.js:118]: Please remove it, or the other `useEffect` which invoke `mediaUploadSync` here :)
https://github.com/WordPress/gutenberg/pull/26727/files#diff-efe4dc27db844b16612461b3acc8ac6d6ee9c44a490baf9bb58c03e6f88b77d7R181

**Review by dratwas — CHANGES_REQUESTED** [review]

**Review by dratwas — APPROVED** [review]

File: packages/components/src/mobile/image/image-editing-button.native.js [hunk 3]
```diff
@@ -27,7 +33,12 @@ const ImageEditingButton = ( {
 			source={ { uri: url } }
 			openReplaceMediaOptions={ openMediaOptions }
 			render={ ( { open, mediaOptions } ) => (
-				<TouchableWithoutFeedback onPress={ open }>
+				<TouchableWithoutFeedback
```
Comment (dcalhoun) [inline packages/components/src/mobile/image/image-editing-button.native.js:36]: The changes in this PR may not actually have the desired impact for addressing wordpress-mobile/gutenberg-mobile#2699, for a couple of reasons I have discovered.

1. This edit image button was rendered _beneath_ the image itself, which means it was not reachable when using voice over. I have [relocated it to address this](https://github.com/WordPress/gutenberg/pull/25810/commits/b07e0bccdefe9b3f778d7cdfa9477f270f578509).
    <details>
    <summary>Unreachable edit button</summary>
    
    unreachable-edit-image button
    
    </details>
2. I believe wordpress-mobile/gutenberg-mobile#2699 was describing the lack of a label for the _add media button_, which is displayed when no cover media has been set. I have [added a label for this button](https://github.com/WordPress/gutenberg/pull/25810/commits/e96e1d40bd7ba01bb99b402ab825b7c06cf4f954).

    <details>
    <summary>Label-less add button</summary>
    
    label-less-add-media-button
    
    https://user-images.githubusercontent.com/438664/107584541-3e6b9f00-6bc2-11eb-89cd-183bbf6d65ee.MP4
    
    </details>

File: packages/components/src/mobile/image/image-editing-button.native.js [hunk 4]
```diff
@@ -27,7 +33,12 @@ const ImageEditingButton = ( {
...
+					onPress={ open }
+					accessibilityRole={ 'button' }
+					accessibilityLabell={ __( 'Edit the cover image' ) }
```
Comment (dcalhoun) [inline packages/components/src/mobile/image/image-editing-button.native.js:39]: It looks like there was a typo in the prop name here. Additionally, I have [changed the label](https://github.com/WordPress/gutenberg/blob/0aa90e57954b507a076d23f9d6cd8d0d7c45a48b/packages/components/src/mobile/image/image-editing-button.native.js#L38) to be `Edit image`, as this button is also used for the standard image block.

**Review by dcalhoun — COMMENTED** [review]
Hey @lukewalczak. 👋🏻 I discovered this PR from needing to resolve merge conflicts with my work in #25810. I had actually made some changes related to a11y without realizing the issues were already documented in an Issue or that you were working to address some of these.

That said, from reviewing your work and resolving the merge conflicts, I  believe I identified a few issues. Will you please review my comments to make sure that my changes are appropriate and do not introduce regressions?

Thanks!

**Pull request merged**